### PR TITLE
Add wranger post prop when wranger makes posts

### PR DESCRIPTION
**Pull request by crspeller**
It would be nice to know posts that have been moved by wrangler programmatically.

One case is https://github.com/mattermost/mattermost-plugin-ai would like to ignore wrangler moved posts.

### CODE REVIEW

File: server/message_helpers.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"github.com/pkg/errors"
 )
 
+const WrangerProp = "wrangler"
```
Comment (gabrieljackson) [inline server/message_helpers.go:11]:
```suggestion
const wrangerProp = "wrangler"
```

File: server/message_helpers.go [hunk 2]
```diff
@@ -244,6 +246,11 @@ func (p *Plugin) createPostWithRetries(post *model.Post, retryDuration time.Dura
 	}
 }
 
+func (p *Plugin) createPostForUser(post *model.Post) (*model.Post, *model.AppError) {
+	post.AddProp(WrangerProp, true)
```
Comment (gabrieljackson) [inline server/message_helpers.go:250]:
```suggestion
	post.AddProp(wrangerProp, true)
```

**Review by gabrieljackson — COMMENTED** [review]
Thanks for kicking this off for me. Do you think we need to add the prop for other messages that wrangler moves or copies in a follow-up PR or does this take care of all of your concerns for the AI plugin?

Looks good other than the one tiny lint issue.

**Comment by crspeller** [comment]
@gabrieljackson In all the other places I see I think wrangler is posting as itself. Those can be identified by the bot name. Unless I am missing some place.

**Comment by gabrieljackson** [comment]
Yeah, this is the big one used for moving or copying threads. https://github.com/gabrieljackson/mattermost-plugin-wrangler/blob/master/server/message_helpers.go#L232

If we want to include the props on those then that should take care of it.

**Comment by crspeller** [comment]
@gabrieljackson Maybe I am misunderstanding. I think the PR already covers that one?

**Comment by gabrieljackson** [comment]
Sorry, you are right. I misread the change originally. This should be good to go. Thanks!

**Pull request merged**